neurom/view/dendrogram.py

```diff
@@ -0,0 +1,176 @@
+        will represent a nice tree structure.
+    '''
+
+    class _PositionedDendrogram(object):
+        '''Wrapper around dendrogram that allows to lay it out.
+
+        The layout happens only in X coordinates. Children's Y coordinate is just a parent's Y
+         + parent's height. Algorithm is that we calculate bounding rectangle width of each
+         dendrogram's subtree. This width is a sum of all children widths calculated recursively
+         in `total_width`. If no children then the width is the dendrogram's width. After the
+         calculation we start to lay out. Each child gets its X coordinate as:
+         parent's X + previous sibling children widths + half of this child's width.
+        '''
+        HORIZONTAL_PADDING = 2
+
+        def __init__(self, dendrogram):
+            self.dendrogram = dendrogram
+            self.children = [_PositionedDendrogram(child) for child in dendrogram.children]
+            self.total_width = self.dendrogram.width
+            if self.children:
+                children_width = np.sum([child.total_width for child in self.children])
+                children_width += self.HORIZONTAL_PADDING * (len(self.children) - 1)
+                self.total_width = max(self.total_width, children_width)
+
+        def position_at(self, origin):  # pylint: disable=missing-docstring
+            positions = {self.dendrogram: origin}
+            if self.children:
+                end_point = origin + [0, self.dendrogram.height]
+                left_bottom_offset = [-.5 * self.total_width, 0]
+                children_origin = end_point + left_bottom_offset
+                for child in self.children:
+                    child_origin = children_origin + [.5 * child.total_width, 0]
+                    positions.update(child.position_at(child_origin))
+                    children_origin += [child.total_width + self.HORIZONTAL_PADDING, 0]
+            return positions
+
+    pos_dendrogram = _PositionedDendrogram(dendrogram)
+    return pos_dendrogram.position_at(origin)
+
+
+def get_size(positions):
+    '''Get the size of bounding rectangle that embodies positions.
+
+    Args:
+        positions (dict of Dendrogram: np.array): positions xy coordinates of dendrograms
+
+    Returns:
+        Tuple of width and height of bounding rectangle.
+    '''
+    max_y_list = [dendrogram.height + coords[1] for dendrogram, coords in positions.items()]
+    coords = np.array(list(positions.values()))
+    width = np.max(coords[:, 0]) - np.min(coords[:, 0])
+    height = np.max(max_y_list) - np.min(coords[:, 1])
+    return width, height
+
+
+def move_positions(positions, to_origin):
+    '''Move positions to a new origin.
+
+    Args:
+        positions (dict of Dendrogram: np.array): positions
+        to_origin (np.array): where to move. np.array of (2,) shape for x,y coordindates.
+
+    Returns:
+        Moved positions.
+    '''
+    to_origin = np.array(to_origin)
```

Comment:
```suggestion
    to_origin = np.asarray(to_origin)
```
will prevent a copy if `to_origin` is already a numpy array.